src/FSharpPlus/Lens.fs

```diff
@@ -112,6 +112,39 @@ module Lens =
     /// Lens for the fifth element of a tuple
     let inline _5 f t = Map.InvokeOnInstance (fun x -> mapItem5 (fun _ -> x) t) (f (item5 t))
 
+    [<RequireQualifiedAccess>]
+    module List=
+        let removeAt i lst =
+            if List.length lst > i then
+                lst.[0..i-1] @ lst.[i+1..]
+            else lst
+
+        let setAt i x lst =
+            if List.length lst > i then
+                lst.[0..i-1] @ x::lst.[i+1..]
+            else lst
+
+        /// Given a specific key, produces a Lens from a List<value> to an Option<value>. When setting,
+        /// a Some(value) will insert or replace the value into the map at the given index. Setting a value of
+        /// None will delete the value at the specified index.  Works well together with non.
+        let inline _item i f t = Map.InvokeOnInstance
+                                  (function | None -> removeAt i t | Some x -> setAt i x t)
+                                  (f (List.tryItem i t))
+
+    [<RequireQualifiedAccess>]
+    module Array=
+        let setAt i x arr =
+            if Array.length arr > i then
+                Array.set arr i x
+                arr
+            else
+                arr
```

Comment:
This should go in Array.fs in Extensions, either that or make ir a private function, the latter option being the safest.

Reply:
Flagging as private - there are already enough ways to set a value in an array